readers/api/transport.go

```diff
@@ -141,6 +143,16 @@ func decodeList(_ context.Context, r *http.Request) (interface{}, error) {
 		return nil, errors.Wrap(apiutil.ErrValidation, err)
 	}
 
+	aggregation, err := apiutil.ReadStringQuery(r, aggregationKey, "")
+	if err != nil {
+		return nil, errors.Wrap(apiutil.ErrValidation, err)
+	}
+
+	interval, err := apiutil.ReadStringQuery(r, intervalKey, "")
```

Comment:
Since interval and aggregation are tightly coupled we can have something like this, where we have a default interval value if the aggregation is present. This can be any value based on the generic usecase.
```go
	var interval string
	if aggregation != "" {
		interval, err = apiutil.ReadStringQuery(r, intervalKey, "1 day")
		if err != nil {
			return nil, errors.Wrap(apiutil.ErrValidation, err)
		}
	}
```

This will make validations simpler since if aggregation is present the interval definitely has a value.

```go
	if req.pageMeta.Aggregation != "" && req.pageMeta.To == 0 {
		return apiutil.ErrMissingTo
	}
```

Reply:
@ianmuchyri what would be the ideal interval?

Reply:
Lets go with the minimum we allow as the default, hence we can go with `1s`